exes/ethercat/inc/public/tfc/ec/devices/schneider/atv320.hpp

```diff
@@ -552,8 +615,17 @@ class atv320 final : public base {
       });
     }
 
-    out->command_word = static_cast<uint16_t>(command);
-    out->frequency = running_ ? static_cast<uint16_t>(reference_frequency_) : 0;
+    if (reference_frequency_.reverse) {
+      std::bitset<16> command_word{ std::to_underlying(command) };
+      static constexpr std::size_t reverse_bit =
+          11;  // from https://iportal2.schneider-electric.com/Contents/docs/SQD-ATV320U11N4C_USER%20GUIDE.PDF
+      command_word.set(reverse_bit, true);
+      out->command_word = static_cast<uint16_t>(command_word.to_ulong());
+    } else {
+      out->command_word = static_cast<uint16_t>(command);
+    }
+
+    out->frequency = running_ ? reference_frequency_.value.numerical_value_ : 0;
```

Comment:
This looks expensive. We have a control_word type bit field that is returned from the cia_402 header. Could we not destroy that type, modify the reserved bit on the bit field and then utilize a std::bit_cast before copying the 16-bit integer to the wire.

Reply:
refactored